x-pack/platform/plugins/shared/streams_app/public/components/streams_graph/streams_graph.tsx

```diff
@@ -0,0 +1,329 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { CSSProperties } from 'react';
+import React, { createContext, memo, useCallback, useEffect, useRef, useState } from 'react';
+import { isEqual } from 'lodash';
+import cytoscape from 'cytoscape';
+import dagre from 'cytoscape-dagre';
+import type { EuiThemeComputed } from '@elastic/eui';
+import { useEuiTheme } from '@elastic/eui';
+import { Streams } from '@kbn/streams-schema';
+import { useStreamsAppRouter } from '../../hooks/use_streams_app_router';
+
+cytoscape.use(dagre);
+
+export interface StreamsGraphProps {
+  streams?: Array<{ stream: Streams.all.Definition }>;
+  currentStream?: Streams.all.Definition;
+}
+
+export function StreamsGraph({ streams, currentStream }: StreamsGraphProps) {
+  const router = useStreamsAppRouter();
+
+  const onNodeClick = useCallback(
+    (id: string) => {
+      router.push('/{key}', {
+        path: { key: id },
+        query: {},
+      });
+    },
+    [router]
+  );
+
+  return <Cytoscape elements={buildGraph(streams, currentStream)} onNodeClick={onNodeClick} />;
+}
+
+function buildGraph(
+  streams?: Array<{ stream: Streams.all.Definition }>,
+  currentStream?: Streams.all.Definition
+): cytoscape.ElementDefinition[] {
+  if (!streams) {
+    return [];
+  }
+
+  if (!currentStream) {
```

Comment:
nit we don't need this special case since currentStream is always defined in the current call chain